### PR TITLE
Properly populate metrics uri in presence of auth failures

### LINKED ISSUE #24938: Micrometer URI templating does not apply for unauthorized requests
### Describe the bug

In case a HTTP request ends up in `401 Unauthorized`, the Micrometer URI tag will not be templated, but include the full URI. If the request with the same URI is correctly authenticated, then the URI will be templated.

### Expected behavior

* URIs are correctly templated no matter what the response code is 

### Actual behavior

* does not work, at least with the `quarkus-resteasy-reactive`

### How to Reproduce?

Easy to reproduce with:

```
@Path("/hello")
public class GreetingResource {

    @GET
    @Path("something/{message}")
    @Produces(MediaType.TEXT_PLAIN)
    public Uni<RestResponse<String>> hello(@PathParam("message") String message) {
        return Uni.createFrom().item(RestResponse.ok(message));
    }
}
```

and:

```
%dev.quarkus.security.users.embedded.enabled=true
%dev.quarkus.security.users.embedded.plain-text=true
%dev.quarkus.security.users.embedded.users.scott=reader
%dev.quarkus.security.users.embedded.users.stuart=writer
%dev.quarkus.security.users.embedded.roles.scott=READER
%dev.quarkus.security.users.embedded.roles.stuart=READER,WRITER

quarkus.http.auth.permission.default.paths=/hello/*
quarkus.http.auth.permission.default.policy=authenticated
```

Results in the following URIs, which provide that `401` requests are not templated and `200` are:

```
http_server_requests_seconds_count{method="GET",outcome="CLIENT_ERROR",status="401",uri="/hello/something/my",} 3.0
http_server_requests_seconds_sum{method="GET",outcome="CLIENT_ERROR",status="401",uri="/hello/something/my",} 0.013654029
http_server_requests_seconds_count{method="GET",outcome="SUCCESS",status="200",uri="/hello/something/{message}",} 1.0
http_server_requests_seconds_sum{method="GET",outcome="SUCCESS",status="200",uri="/hello/something/{message}",} 0.029346689
http_server_requests_seconds_count{method="GET",outcome="CLIENT_ERROR",status="401",uri="/hello/something/asdsad",} 3.0
http_server_requests_seconds_sum{method="GET",outcome="CLIENT_ERROR",status="401",uri="/hello/something/asdsad",} 0.535294234
```

### Output of `uname -a` or `ver`

Linux ise-Precision-5550 5.13.0-39-generic #44~20.04.1-Ubuntu SMP Thu Mar 24 16:43:35 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537) Maven home: /home/ise/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4 Java version: 17.0.2, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64 Default locale: en_US, platform encoding: UTF-8 OS name: "linux", version: "5.13.0-39-generic", arch: "amd64", family: "unix"

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
We now are able for RESTEasy Reactive to determine the URI template even when auth failures occur

Fixes: #24938

### CODE REVIEW

**Review by ebullient — APPROVED** [review]

**Comment by geoand** [comment]
I'll add some tests for this tomorrow

**Comment by geoand** [comment]
PR updated with tests for both Micrometer and OpenTelemetry

**Comment by geoand** [comment]
@brunobat is there a specific OpenTelemetry issue I should link to in this PR or does https://github.com/quarkusio/quarkus/issues/24938 cover both?

**Comment by geoand** [comment]
I believe we can backport this one. WDYT?

File: integration-tests/opentelemetry-reactive/pom.xml [hunk 1]
```diff
@@ -41,6 +41,12 @@
             <artifactId>opentelemetry-sdk-testing</artifactId>
         </dependency>
 
+        <!-- Security -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-elytron-security-properties-file</artifactId>
```
Comment (brunobat) [inline integration-tests/opentelemetry-reactive/pom.xml:47]: Why do we need this now, @geoand ?

Comment (brunobat) [reply integration-tests/opentelemetry-reactive/pom.xml:47]: got it. You are adding security to simulate early errors.

Comment (geoand) [reply integration-tests/opentelemetry-reactive/pom.xml:47]: Exactly - this was the easiest way to do that

**Comment by michalsomora** [comment]
Hi, is there a any chance this fix will be downmerged to quarkus 2?

**Comment by brunobat** [comment]
> Hi, is there a any chance this fix will be downmerged to quarkus 2?

This won't be backported.

**Pull request merged**